### ISSUE TITLE
MS73 layer models might be overfitting

### ISSUE THREAD

**Issue by JoyfulGen** [DDMAL/Rodan#1202]
I can best explain this with a scenario:

I did a Pixel run with folio 023. I then produced models with that run and used those to do a second Pixel run with folio 017. Those latest models could separate folio 017 fairly well. I then did a third Pixel run with folio 260. Those models separated folio 260 perfectly, but when they tried to separate 017 they produced three almost completely empty layers.

This has been happening fairly often; models we make do a great job separating the folio they were just trained on, but can't separate any other folio, including ones they were trained on previously.

These are the models I'm talking about:
[Background Model MTWS 023 round 3.hdf5.zip](https://github.com/user-attachments/files/16893414/Background.Model.MTWS.023.round.3.hdf5.zip)
[Model 1 MTWS 023 round 3.hdf5.zip](https://github.com/user-attachments/files/16893416/Model.1.MTWS.023.round.3.hdf5.zip)
[Model 2 MTWS 023 round 3.hdf5.zip](https://github.com/user-attachments/files/16893418/Model.2.MTWS.023.round.3.hdf5.zip)
[Model 3 MTWS 023 round 3.hdf5.zip](https://github.com/user-attachments/files/16893424/Model.3.MTWS.023.round.3.hdf5.zip)

@kyrieb-ekat did I explain that right

**Comment by kyrieb-ekat** [comment]
Yes! What we're finding is that the models trained on specific images will do well generalizing to new random things we show them, and then at some point completely stop or freak out; usually in the vein of #1204 where we're seeing a layer stop being produced entirely, or layers are combined (usually text and music).

**Comment by fujinaga** [comment]
Does the freak out occur without retraining? What happens if you try the second time w/o retraining?

**Comment by JoyfulGen** [comment]
@fujinaga so far the models I've been creating have been producing consistent results; if I run the same folio twice with the same layer separation models, the results are exactly the same. So I believe the freak out happens after retraining. @kyrieb-ekat did mention something weird happening when using models trained in staging on a Paco Classifying job in production (Kyrie I'm sorry I can't remember the details!)

**Comment by kyrieb-ekat** [comment]
Question: in the iterating models were the previous models used as inputs in the training run, or just the ZIP files? It makes sense more labeled examples of various items would provide more information on how to handle them when they show up. 

This abuts nicely with one thing I've been testing in the background about at what point rodan does best with too many samples (like, is it better to have ALL the samples added with each successive correction and not cycle them out, or is it better to retain like, three and cycle out the oldest ones), where Rodan does better with the previous training models as inputs and at least two sample inputs. There may or may not be a memory ceiling (so to speak) with samples, but since you were able to manage 5 without it breaking we might be able to get away with more than we think.

**Comment by JoyfulGen** [comment]
Just the ZIP files! Wait... you can input models to the Paco Trainer job? Can't you just import layers?

**Comment by kyrieb-ekat** [comment]
Yes! You can add input ports for previous models. I'm finding in my testing that at least three samples and previous training models produce best results. 

<img width="416" alt="Screenshot 2024-09-10 at 11 27 11 AM" src="https://github.com/user-attachments/assets/c4a17a1e-3168-4cf7-9a22-6c672f03c581">

I'm working on testing a single model on a variety of images to see if I can consistently replicate the weird "tiles" phenomena. Apparently there's no memory impact by having many input samples, and since Rodan doesn't remember previous runs we don't need to worry about reusing the same samples over and over again, we can show it as many samples in a run as we like (hopefully!).

**Comment by JoyfulGen** [comment]
UPDATE: I have conducted yet another test and obtained yet another result.

This time I tried making models using patches of a single folio. I ran the folio in Pixel four times, using a different chunk of folio 077 each time (the four chunks added up to the complete folio). The first set of models I made like so:

- Correct each patch in Pixel from nothing;
- Chuck all four ZIP files into Paco Trainer at the same time (no iteration).

The second I made like so:

- Made models from the two first patches;
- Corrected patch 3 using those models;
- Made models from patch 3, as well as the first two patches;
- Corrected patch 4 using those models;
- Made models from patch 4, as well as patch 3 and 1+2 (yes iteration).

First of all, neither set of models work particularly well. Both still get layers 1 and 3 pretty mixed up. Am I just not using enough samples? 

Second of all, this time the iteration models did slightly better. The main difference is again in the border: the iteration models don't include any borders almost at all, whereas the combo models have a huge amount of border, usually in layer 1.

Demo: these are the layer 1's of folio 055, with the combo models on the left and the iteration models on the right.
<img width="890" alt="055 layer 1's side by side" src="https://github.com/user-attachments/assets/85394ece-8f4b-4af2-bcc8-cbb4e0fb312e">

Why is the border treated so differently when I painted over the exact same things? And why is this result the opposite of the previous one, where the iteration model is the one that included the border? @fujinaga is this expected behaviour? I would be grateful for futher guidance, because I'm a bit at loose ends at this point.

**Comment by kyrieb-ekat** [comment]
I'm going to leave this open, because the border behavior is still present, though we were able to ultimately figure out a lot of the other things with the MS73 Strategy/Approach, which is delineated in detail in #1204 

It might be worth, however, opening a distinct separate issue about the border behavior, in terms of experimenting with how to remove it or identifying if it is part of the larger layer separation thing we may be experiencing at the moment? Otherwise, for now, I think leaving this open might be the move.... @JoyfulGen I'd love your thoughts.
